templates/primary-secondary/test/primary-secondary.visual-diff.js

```diff
@@ -0,0 +1,33 @@
+const puppeteer = require('puppeteer');
+const VisualDiff = require('@brightspace-ui/visual-diff');
+
+describe('d2l-template-primary-secondary', function() {
+	const visualDiff = new VisualDiff('primary-secondary', __dirname);
+	let browser;
+	let page;
+
+	before(async() => {
+		browser = await puppeteer.launch();
+		page = await browser.newPage();
+	});
+
+	after(() => browser.close());
+
+	describe('desktop', function() {
+		before(async() => {
+			await page.setViewport({width: 930, height: 930, deviceScaleFactor: 2});
+		});
+
+		[
+			{ name: 'normal', fileName: 'primary-secondary-desktop.visual-diff.html' },
+			{ name: 'larger-than-viewport-height', fileName: 'primary-secondary-desktop-large.visual-diff.html' }
+		].forEach((entry) => {
+			it(entry.name, async function() {
+				await page.goto(`${visualDiff.getBaseUrl()}/templates/primary-secondary/test/${entry.fileName}`, {waitUntil: ['networkidle0', 'load']});
+				await page.bringToFront();
+				const rect = await visualDiff.getRect(page, 'd2l-template-primary-secondary');

```

Comment:
Note: the API for `getRect` accepts `(page, selector, margin)`, the `margin` being optional and defaulting to `10px`. The reason for the default margin is that many of our components use `box-shadow` for various purposes, but `box-shadow` falls outside of the bounding rect as measured by the browser. In order for us to reliably capture the shadow, we default to taking a screenshot slightly larger than the component, but this may be overridden as necessary. This is also why we try to separate the fixtures (when there are multiple on a single page)... so that a screenshot does not include the bottom/top of adjacent fixtures.  Anyway, that's the background... long story short, we _might_ want `0` for the margin in this case. Not sure if it makes a difference in this case - it might, but we might not have noticed.